### PR TITLE
update component-cli lib and simplify ctf build

### PR DESCRIPTION
**Pull request by schrodit**
**How to categorize this PR?**

/area delivery
/priority 3

**What this PR does / why we need it**:

Simplify the landscaper artifact generation within the component descriptor generation

also added a local script to test the ctf generation

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: hack/generate-cd.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+if ! which component-cli 1>/dev/null; then
+  echo -n "component-cli is required to generate the component descriptors"
+  echo -n "Trying to installing it..."
+  go get github.com/gardener/component-cli/cmd/component-cli
+
+  if ! which component-cli 1>/dev/null; then
+    echo -n "component-cli was successfully installed but the binary cannot be found"
+    echo -n "Try adding the \$GOPATH/bin to your \$PATH..."
+    go get github.com/gardener/component-cli/cmd/component-cli
```
Comment (MrBatschner) [inline hack/generate-cd.sh:27]: If $GOPATH is not in $PATH, running `go get` a second time probably will not help, instead, we should exit.

File: hack/generate-cd.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+#access:
+#  type: 'ociRegistry'
+#  imageReference: ${image}
+#EOF
+#done
+
+echo "> Creating ctf"
+CTF_PATH=./gen/ctf.tar
+mkdir -p ./gen
+rm ${CTF_PATH}
```
Comment (MrBatschner) [inline hack/generate-cd.sh:60]: if `./gen/ctf.tar` did not exist and with `set -e` set, the script will exit here
```suggestion
[ -e $CTF_PATH ] && rm ${CTF_PATH}
```

**Review by MrBatschner — COMMENTED** [review]
Just two minor things... and something does not seem to be ok with building the component-descriptors, it fails in the publish step - see below.

File: .ci/component_descriptor [hunk 3]
```diff
@@ -23,21 +23,12 @@ function buildComponentArchive() {
   CA_PATH="$(mktemp -d)"
   printf "> Building component ${COMPONENT_NAME}\n"
 
-  component-cli ca resources add ${CA_PATH} \
+  component-cli ca "${CA_PATH}" "${CTF_PATH}" \
```
Comment (MrBatschner) [inline .ci/component_descriptor:26]:
```suggestion
  component-cli ca create "${CA_PATH}" "${CTF_PATH}" \
```

Comment (schrodit) [reply .ci/component_descriptor:26]: this is right. but the we need to wait for the new component-cli version to the reach concourse.

**Pull request merged**